cmip6_downscaling/methods/deepsd/tasks.py

```diff
@@ -0,0 +1,475 @@
+    """
+    ds_hash = str_to_hash(str(source_path) + str(obs_path))
+    target = results_dir / 'deepsd_rescale' / ds_hash
+
+    if use_cache and zmetadata_exists(target):
+        print(f'found existing target: {target}')
+        rescaled_ds = xr.open_zarr(target)
+        return target
+
+    orig_ds = xr.open_zarr(source_path)
+    obs_ds = xr.open_zarr(obs_path).sel(
+        time=slice(run_parameters.train_dates[0], run_parameters.train_dates[1])
+    )
+    obs_mean = obs_ds.mean(dim='time')
+    obs_std = obs_ds.std(dim='time')
+
+    rescaled_ds = (orig_ds * (obs_std[run_parameters.variable] + EPSILON)) + obs_mean[
+        run_parameters.variable
+    ]
+    # Clip negative precipitation values
+    if run_parameters.variable == "pr":
+        rescaled_ds = rescaled_ds.clip(min=0)
+    rescaled_ds.attrs.update({'title': 'deepsd_output'}, **get_cf_global_attrs(version=version))
+    print(f'writing rescaled dataset to {target}')
+    rescaled_ds.pipe(set_zarr_encoding).to_zarr(target, mode='w')
+    return target
+
+
+@task(log_stdout=True)
+def normalize_gcm(predict_path: UPath, historical_path: UPath) -> UPath:
+    """Normalize gcm data based on historical data.
+
+    Parameters
+    ----------
+    predict_path : UPath
+        Path to dataset that will be normalized.
+    historical_path : UPath
+        Path to dataset to normalized based on.
+    Returns
+    -------
+    UPath
+        Path to normalized dataset.
+    """
+    # Create path for output file
+    ds_hash = str_to_hash(str(predict_path) + str(historical_path))
+    target = intermediate_dir / 'normalize' / ds_hash
+
+    # Skip step if output file already exists when using cache
+    if use_cache and zmetadata_exists(target):
+        print(f'found existing target: {target}')
+        norm_ds = xr.open_zarr(target)
+        return target
+
+    predict_ds = xr.open_zarr(predict_path)
+    historical_ds = xr.open_zarr(historical_path)
+
+    historical_ds_mean = historical_ds.mean(dim="time").compute()
+    historical_ds_std = historical_ds.std(dim="time").compute()
+
+    norm_ds = (predict_ds - historical_ds_mean) / (historical_ds_std + EPSILON)
+
+    norm_ds = lon_to_180(norm_ds)
+    norm_ds = norm_ds.chunk({'time': INFERENCE_BATCH_SIZE, 'lat': -1, 'lon': -1})
+
+    norm_ds.attrs.update({'title': 'normalize'}, **get_cf_global_attrs(version=version))
+    print(f'writing normalized predict dataset to {target}')
+    norm_ds.pipe(set_zarr_encoding).to_zarr(target, mode='w')
+
+    return target
+
+
+@task(log_stdout=True)
+def inference(gcm_path: UPath, run_parameters: RunParameters) -> UPath:
+    """Run inference on normalized gcm data.
+
+    Parameters
+    ----------
+    gcm_path : UPath
+        Path to normalized dataset.
+    run_parameters : RunParameters
+        Parameters for run set-up and model specs.
+    Returns
+    -------
+    UPath
+        Path to dataset containing model predictions.
+    """
+    import tensorflow_io
```

Comment:
adding `# noqa` should suffice for flake8 (pre-commit) to ignore this line, 

```suggestion
    import tensorflow_io # noqa
```